### DISCUSSION TITLE
Question: Use nuclei as a library for thrid-party projects?

### DISCUSSION THREAD

**Discussion by dubs3c** [projectdiscovery/nuclei#546] [Q&A] [answered]
It does not seem like nuclei was designed to be used as a library that people can use in their own golang projects. I guess you could programmatically set the flag options but that feels more like a hack.

Maybe I'm wrong though?

**Comment by Ice3man543** [comment, marked as answer]
@dubs3c this should be possible with the new nuclei refactor currently in dev branch. Initially nuclei wasn't written with Embedded use cases In mind, the new release should fix that to a certain extent! https://github.com/projectdiscovery/nuclei/tree/dev. We will add more usage examples with the new release.

**Reply by dubs3c** [reply]
Alright, sounds good :) Do you know how the API will look like? Or is it still in active development?

**Comment by jsgv** [comment]
Seems like this is still not possible. 
I believe it's the way that Go handles the special `internal` directory. 
[Dave Cheney blog post](https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface) describing some of the behaviors of `internal` folder.

![golang](https://user-images.githubusercontent.com/2798097/118281815-ec425e80-b4cd-11eb-92a6-f7d861d51de8.jpg)

**Reply by dubs3c** [reply]
Oh yeah, that is true :/

**Reply by jsgv** [reply]
It might be as easy as renaming the folder and the references to it, but I am not familiar with inner workings of Nuclei.
